ci-operator/jobs/image-mirroring.yaml

```diff
@@ -0,0 +1,290 @@
+            - |
+              set -o errexit
+              failures=0
+              for mapping in /etc/imagemirror/mapping_knative*; do
+                echo "Running: oc image mirror -f=$mapping --skip-multiple-scopes"
+                if ! oc image mirror -f="$mapping" --skip-multiple-scopes; then
+                  echo "ERROR: Failed to mirror images from $mapping"
+                  failures=$((failures+1))
+                fi
+              done
+              exit $failures
+          command:
+            - /bin/bash
+          env:
+            - name: HOME
+              value: /home/mirror
+          image: "quay.io/openshift/origin-cli:4.1"
+          imagePullPolicy: Always
+          name: ""
+          resources:
+            requests:
+              cpu: 500m
+          volumeMounts:
+            - mountPath: /home/mirror/.docker/config.json
+              name: push
+              readOnly: true
+              subPath: .dockerconfigjson
+            - mountPath: /etc/imagemirror
+              name: config
+      volumes:
+        - name: push
+          secret:
+            secretName: registry-push-credentials-quay.io-openshift-knative
+        - name: config
+          configMap:
+            name: image-mirror-mappings
+  - agent: kubernetes
+    labels:
+      ci.openshift.io/role: image-mirroring
+      ci.openshift.io/area: openshift
+    cron: "@midnight"
+    decorate: true
+    name: periodic-image-mirroring-openshift
+    skip_cloning: true
+    spec:
+      containers:
+        - args:
+            - -c
+            - |
+              set -o errexit
+              failures=0
+              for mapping in /etc/imagemirror/mapping_origin*; do
+                echo "Running: oc image mirror -f=$mapping --skip-multiple-scopes"
+                if ! oc image mirror -f="$mapping" --skip-multiple-scopes; then
+                  echo "ERROR: Failed to mirror images from $mapping"
+                  failures=$((failures+1))
+                fi
+              done
+              exit $failures
+          command:
+            - /bin/bash
+          env:
+            - name: HOME
+              value: /home/mirror
+          image: "quay.io/openshift/origin-cli:4.1"
+          imagePullPolicy: Always
+          name: ""
+          resources:
+            requests:
+              cpu: 500m
+          volumeMounts:
+            - mountPath: /home/mirror/.docker/config.json
+              name: push
+              readOnly: true
+              subPath: .dockerconfigjson
+            - mountPath: /etc/imagemirror
+              name: config
+      volumes:
+        - name: push
+          secret:
+            secretName: registry-push
+        - name: config
+          configMap:
+            name: image-mirror-mappings
+  - agent: kubernetes
+    labels:
+      ci.openshift.io/role: image-mirroring
+      ci.openshift.io/area: tekton
+    cron: "@midnight"
+    decorate: true
+    name: periodic-image-mirroring-tekton
```

Comment:
Isn't tekton part of knative?

Reply:
I'm not sure about their relation, but they set up mirroring config separately (I have just followed the existing jobs)

Reply:
weird